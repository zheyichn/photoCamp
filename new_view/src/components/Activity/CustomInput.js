import React, { useState } from "react";
import { Grid, TextField } from "@mui/material";
import ConfirmButton from "../Button/ConfirmButton";
import CancelButton from "../Button/CancelButton";

export default function CustomInput(props) {
  const { defaultValue, saveText, cancelEdit } = props;
  const [value, setValue] = useState(defaultValue);
  const [disableConfirm, setDisableComfirm] = useState(true);

  const updateText = () => {
    saveText(value);
  };

  return (
    <>
      <Grid flex-direction="row" container>
        <Grid item width="100%">
          <TextField
            fullWidth
            onChange={(e) => {
              if (e.target.value != defaultValue && e.target.value != "") {
                setDisableComfirm(false);
              }
              setValue(e.target.value);
            }}
            value={value}
          />
        </Grid>
        <Grid item container mt={1} flexDirection="row">
          <Grid mr={2}>
            <ConfirmButton disabled={disableConfirm} onClick={updateText} />
          </Grid>
          <Grid>
            <CancelButton pl={2} onClick={cancelEdit}></CancelButton>
          </Grid>
        </Grid>
      </Grid>
    </>
  );
}
