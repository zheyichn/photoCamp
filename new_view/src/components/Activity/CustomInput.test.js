import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import CustomInput from "./CustomInput";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";

afterEach(() => {
  cleanup();
});

describe("CustomInput component renders correctly", () => {
  test("buttons displayed correctly", () => {
    render(<CustomInput />);
    const confirmButton = screen.getByTestId("confirm-btn");
    const cancelButton = screen.getByTestId("cancel-btn");
    expect(confirmButton).toBeInTheDocument();
    expect(cancelButton).toBeInTheDocument();
  });

  it("customInput match snapshots", () => {
    const tree = renderer.create(<CustomInput />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
