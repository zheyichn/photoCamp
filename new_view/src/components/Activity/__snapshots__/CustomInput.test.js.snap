// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput component renders correctly customInput match snapshots 1`] = `
<div
  className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  flex-direction="row"
>
  <div
    className="MuiGrid-root MuiGrid-item css-1mum6ye-MuiGrid-root"
  >
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r0:"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-ihdtdm"
          >
            <span
              className="notranslate"
            >
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item css-1shgmd9-MuiGrid-root"
  >
    <div
      className="MuiGrid-root css-6ienvx-MuiGrid-root"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1fxqi4r-MuiButtonBase-root-MuiButton-root"
        data-testid="confirm-btn"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        Confirm
      </button>
    </div>
    <div
      className="MuiGrid-root css-vj1n65-MuiGrid-root"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-120lcjt-MuiButtonBase-root-MuiButton-root"
        data-testid="cancel-btn"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;
